Cookbook/Technical-Documents/BackendFeatureHowTo.md

```diff
@@ -0,0 +1,204 @@
+# How to implement a feature that works with the backend
+Let's look at a minimal and simple example of working with the backend to get some items into our view model, before moving on to more difficult topics.
+
+## A simple example
+Imagine some endpoint on the backend returns this piece of JSON:
+``` json
+{
+  "cities": [
+    {
+      "name": "London"
+    },
+    {
+      "name": "Bath"
+    }
+  ]
+}
+```
+
+First, we want to define some data types, to which we will map this JSON. This is known as a *response*.
+
+### Response
+A `Response` is a data type which wraps the whole response returned from a network call. In many cases it is unnecessary and you can extract the values from a single key directly, just like in this example, where there is only one key. For the sake of this exercise, let's stick with it though.
+
+A response data type is a plain struct conforming to `Equatable`  (this is usually necessary for tests) and `Decodable`. We also define an additional data type for the city itself. You will want to define all the new data types necessary to describe a response:
+``` swift
+struct CitiesResponse: Equatable, Decodable {
```

Comment:
You don't need that wrapper. In the BackendResource you can put the response handler with `Parser.decode(Parser.parse(data, key: "cities")` inside.

I know you have written that for the sake of this exercise you choose to stick with it, however it's something I would mark on PR to have less types if possible.

Reply:
I say that in the previous paragraph :) I will make an example of how to invoke `Parser` with a subkey later.